debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresStreamingChangeEventSource.java

```diff
@@ -78,6 +79,9 @@ public PostgresStreamingChangeEventSource(PostgresConnectorConfig connectorConfi
         this.clock = clock;
         this.schema = schema;
         this.offsetContext = (offsetContext != null) ? offsetContext : PostgresOffsetContext.initialContext(connectorConfig, connection, clock);
+        // replication slot could exist at the time of starting Debezium so we will stream from the position in the slot
+        // instead of the last position in the database
+        this.hasStartLsnStoredInContext = (offsetContext != null);
```

Comment:
What about the changes I propose in:
https://github.com/jpechane/debezium/pull/3/files

Reply:
I like your approach, and I believe it will fix the bug. But it seems a little dangerous to keep the initialisation to latest txn here
```
PostgresOffsetContext.initialContext(connectorConfig, connection, clock);
```

Reply:
@cjbooms - could you please open the PR against Debezium repo not my repo?
Thanks

Reply:
The PR is a suggested commit against your branch.

Reply:
@jpechane Here it is. I'll let you decide in the morning which one to close
https://github.com/debezium/debezium/pull/1325/files